.cbmc-batch/source/make_common_data_structures.c

```diff
@@ -152,3 +152,10 @@ void ensure_cryptosdk_keyring_has_allocated_members(
     keyring->refcount.value = can_fail_malloc(sizeof(size_t));
     keyring->vtable         = nondet_bool() ? NULL : vtable;
 }
+
+void ensure_vtable_has_allocated_members(struct aws_cryptosdk_keyring_vt *vtable, size_t max_len) {
+    if (vtable) {
+        vtable->name = ensure_c_str_is_allocated(max_len);
+        __CPROVER_assume(aws_c_string_is_valid(vtable->name));
+    }
+}
```

Comment:
I'd rename this function to `nondet_allocate_vtable_members` and move the assumption to a validity function (e.g., `aws_cryptosdk_keyring_vt_is_valid`). Here an example [ensure_c_str_is_allocated](https://github.com/awslabs/aws-c-common/blob/e4d885544ae1f80a2e802d7779e3e754e5562fd0/verification/cbmc/sources/make_common_data_structures.c#L191).

Reply:
Updated!